docs/reference/storybook-deploys-with-circleci.md

```diff
@@ -0,0 +1,27 @@
+---
+title: Storybook Deploys with CircleCI
```

Comment:
We can probably remove a big chunk of [this section](https://github.com/mozilla/fxa/#storybook-in-circleci) on our root readme to not maintain two sets of docs. I think we can probably leave a few sentences and cross-link here for more, the screenshot on the repo home page is helpful IMO, but I'm fine with whatever.